lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -53,16 +56,17 @@ defmodule LivebookWeb.SessionLive.SecretsComponent do
               <.secret_with_badge
                 :for={{secret_name, _} <- Enum.sort(@secrets)}
                 secret_name={secret_name}
-                secret_origin={:session}
+                secret_hub_id={:session}
                 stored="Session"
                 active={secret_name == @prefill_secret_name}
                 target={@myself}
               />
               <.secret_with_badge
                 :for={secret <- @saved_secrets}
+                :if={!(secret.name in Map.keys(@secrets) and @secrets[secret.name] != secret.value)}
                 secret_name={secret.name}
-                secret_origin={secret.origin}
-                stored={stored(secret, @hubs)}
+                secret_hub_id={secret.hub_id}
```

Comment:
I don't think we need to pass this, right? We will just store in the hub stored in assigns.